src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2689,6 +2689,270 @@ var projectUniqueNamesForBuildIntToUpdate
                 installationContext);
         }
 
+        /// <summary>
+        /// Return list of Resolved actions after running preview (without commit) in parallel for buildIntegrated projects.
+        /// </summary>
+        public async Task<IEnumerable<ResolvedAction>> PreviewBuildIntegratedProjectsActionsAsync(
```

Comment:
I just discovered that this method seems to be a slight modification of the existing `PreviewBuildIntegratedProjectActionsAsync` method. The method, being about 250 lines long, contains non-trivial business logic, so I'm a bit concerned that any future changes to either method need the same change applied to the other one, but it will not be obvious to whoever makes the change that this will be needed, potentially leading to bugs.

If the existing `PreviewBuildIntegratedProjectActionsAsync` is no longer used, mark it as obsolete. Otherwise, is there a way to extract common code into methods they both use, or have one of them call the other one?

Reply:
I refactored the code so they same logic. After refactoring code `PreviewBuildIntegratedProjectActionsAsync` and `PreviewBuildIntegratedProjectActionsAsync` share same logic via `PreviewBuildIntegratedProjectsActionsAsync`
method. Please check.

Reply:
Did you forget to push?  I still see a ~250 line implementation of the old, existing method and a ~250 line new method which is almost the same, just with a few things done in a different order.

Since the new method allows multiple projects to be passed in, that makes the old method that only accepts a single project a special case. Therefore, can you delete the entire body of the old method and make it call the new method, so there is only one implemention of this method body?

If you do it in a way such that the git diff shows that most of teh changes are only whitespace, that gives a signal to people reviewing your changes that you haven't changed the business logic and we can be more lenient when reviewing. The current diff makes it look like you wrote all the business logic from scratch and therefore we need to verify all your code to make sure the business logic is correct, making the code review much more difficult and time consuming.  GitHub has an option to hide whitespaces differences. If I use this and all the complex business logic disappears from the diff, that makes the code review much easier.